### PR TITLE
import circuit and tests

### PR DESCRIPTION
**Pull request by zjma**
## Description
```
cargo test -p aptos-keyless-circuit-tests
```
- Importing circuit repo from `michael/japanese_characters` branch [link](https://github.com/aptos-labs/aptos-keyless-circuit/commits/michael/japanese_characters) at rev `e633d6d66638b8de7d4a0d4c6953a4ce8201b54c`
- Borrwing a few prover service pieces from `rex/update_tests` branch [link](https://github.com/aptos-labs/prover-service/tree/rex/update_tests) at rev `b06446a9feed2734674c52b8722dc93f858eaa6b`

## Type of Change
- [x] New feature
- [x] Tests

## Which Components or Systems Does This Change Impact?
- [x] keyless circuit

## Key Areas to Review
CI

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: keyless/circuit-data/create-proofs-for-testing.sh [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            echo
+            echo "Cloning or updating aptos-keyless-circuit:$branch in $keylessrepo..."
+            if [ -d $keylessrepo ]; then 
+                echo "Looks like repo is cloned. Updating..."
+                pushd $keylessrepo; git checkout $branch; popd
+            else
+                echo "Cloning..."
+                git clone git@github.com:aptos-labs/aptos-keyless-circuit $keylessrepo
+                pushd $keylessrepo; git checkout $branch; popd
+            fi
```
Comment (zjma) [inline keyless/circuit-data/create-proofs-for-testing.sh:68]: @alinush why was the non-local mode introduced?
We can delete it now, right?

File: keyless/circuit-data/trusted-setup.sh [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            echo
+            echo "Cloning or updating aptos-keyless-circuit:$branch in $keylessrepo..."
+            if [ -d $keylessrepo ]; then 
+                echo "Looks like repo is cloned. Updating..."
+                pushd $keylessrepo; git checkout $branch; popd
+            else
+                echo "Cloning..."
+                git clone git@github.com:aptos-labs/aptos-keyless-circuit $keylessrepo
+                pushd $keylessrepo; git checkout $branch; popd
+            fi
```
Comment (zjma) [inline keyless/circuit-data/trusted-setup.sh:70]: @alinush same Q...

File: keyless/circuit-data/create-proofs-for-testing.sh (hunk 1 above)
Comment (zjma) [reply keyless/circuit-data/create-proofs-for-testing.sh:68]: + @mstraka100

File: keyless/circuit-data/trusted-setup.sh (hunk 2 above)
Comment (zjma) [reply keyless/circuit-data/trusted-setup.sh:70]: + @mstraka100

File: keyless/circuit-data/templates/helpers/sha.circom [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            calcTotal[k].nums[i] <== eqs[i] * sha256compression[i].out[k];
+        }
+        
+        out[k] <== calcTotal[k].sum;
+    }
+}
+
+// Verifies SHA2_256 input padding according to https://www.rfc-editor.org/rfc/rfc4634.html#section-4.1
+template Sha2PaddingVerify(maxInputLen) {
+    signal input in[maxInputLen]; // bit array
```
Comment (zjma) [inline keyless/circuit-data/templates/helpers/sha.circom:83]: actually byte array, right? @mstraka100 @alinush i

Comment (alinush) [reply keyless/circuit-data/templates/helpers/sha.circom:83]: 🤷 not sure... didn't write this.

**Pull request closed without merging**